File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/Staff.swift
import Foundation

// MARK: - Staff (Personal / Colaboradores)

/// Un colaborador registrado en la libreta de personal del organizador.
/// Se asigna a uno o mas eventos via `EventStaff`. Phase 1 no tiene gating
/// por plan — todos los planes pueden crear/editar/eliminar staff.
public struct Staff: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let userId: String
    public var name: String
    public var roleLabel: String?
    public var phone: String?
    public var email: String?
    public var notes: String?
    /// Si esta en `true`, al asignar al staff a un evento se enviara un email
    /// de aviso. La logica real de envio vive en Phase 2 (Pro+).
    public var notificationEmailOptIn: Bool
    /// Hook Phase 3 — poblado cuando el organizer invita al colaborador a
    /// registrarse (solo tier Business). Phase 1 siempre es `nil`.
    public var invitedUserId: String?
    /// Estado de la invitación activa cuando existe (`pending`, `accepted`,
    /// `revoked`, `expired`).
    public var inviteStatus: String?
    public let createdAt: String
    public let updatedAt: String

    // MARK: - Init

    public init(
        id: String,
        userId: String,
        name: String,
        roleLabel: String? = nil,
        phone: String? = nil,
        email: String? = nil,
        notes: String? = nil,
        notificationEmailOptIn: Bool = false,
        invitedUserId: String? = nil,
        inviteStatus: String? = nil,
        createdAt: String,
        updatedAt: String
    ) {
        self.id = id
        self.userId = userId
        self.name = name
        self.roleLabel = roleLabel
        self.phone = phone
        self.email = email
        self.notes = notes
        self.notificationEmailOptIn = notificationEmailOptIn
        self.invitedUserId = invitedUserId
        self.inviteStatus = inviteStatus
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/Staff.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Colaborador / personal recurrente del organizador.
 *
 * Phase 1 — CRUD local del catálogo. El costo NO vive acá: se cobra por
 * asignación en [EventStaff.feeAmount] porque el mismo colaborador puede
 * cobrar distinto en cada evento.
 *
 * Phase 2 activará `notificationEmailOptIn` (email automático al asignarlo a
 * un evento, feature Pro+). En Phase 1 el toggle se guarda pero no dispara
 * nada.
 *
 * Phase 3 (Business) activará [invitedUserId] cuando el organizer invite al
 * colaborador a crear una cuenta. En Phase 1 siempre es null.
 */
@Serializable
data class Staff(
    val id: String = "",
    @SerialName("user_id") val userId: String = "",
    val name: String,
    @SerialName("role_label") val roleLabel: String? = null,
    val phone: String? = null,
    val email: String? = null,
    val notes: String? = null,
    @SerialName("notification_email_opt_in") val notificationEmailOptIn: Boolean = false,
    @SerialName("invited_user_id") val invitedUserId: String? = null,
    @SerialName("invite_status") val inviteStatus: String? = null,
    @SerialName("created_at") val createdAt: String = "",
    @SerialName("updated_at") val updatedAt: String = ""
)
